webapp/components/suggestion/at_mention_provider.jsx

```diff
@@ -99,31 +100,7 @@ class AtMentionSuggestion extends Suggestion {
     }
 }
 
-function filterUsersByPrefix(users, prefix, limit, type) {
-    const filtered = [];
-
-    for (const id of Object.keys(users)) {
-        if (filtered.length >= limit) {
-            break;
-        }
-
-        const user = users[id];
-
-        if (user.delete_at > 0) {
-            continue;
-        }
-
-        if (user.username.startsWith(prefix) ||
-            (user.first_name && user.first_name.toLowerCase().startsWith(prefix)) ||
-            (user.last_name && user.last_name.toLowerCase().startsWith(prefix)) ||
-            (user.nickname && user.nickname.toLowerCase().startsWith(prefix))) {
-            // create a new object here since we're mutating it by adding the type field
-            filtered.push(Object.assign({}, user, {type}));
-        }
-    }
-
-    return filtered;
-}
+let timeoutId = '';
```

Comment:
Can you make `timeoutId` a field of the `AtMentionProvider`? I know it's unlikely that we'll have multiple `AtMentionProvider`s querying the server at once, but it makes me leery having it not encapsulated in the object itself.